NPS Survey: Styling

mattsherman opened the pull request:
This PR styles the NPS Survey dialog.

![nps survey notice](https://cloud.githubusercontent.com/assets/2098816/24615849/75dd12a0-185d-11e7-95aa-4e8d5dcf6e20.gif)

Test:
- To see debug messages, in browse console:
    - `localStorage.setItem('debug', 'calypso:nps-survey');`
- from `http://calypso.localhost:3000/devdocs/blocks/nps-survey`
    - Survey should submit successfully the first time you answer it (the API errors if a score has been provided in the past 4 months)
- from global notice (see #12522)
    - Survey should error (silently; it always will thank the user for submitting -- no need to trouble the user should there be a problem submitting), since the survey name is not yet whitelisted in the API endpoint
- Verify that dismissing survey without answering it calls API (look at debug messages and Redux actions/state)

For now, this is not enabled in any environments. To enable:

```
ENABLE_FEATURES=nps-survey/notice make run
```

To re-trigger the NPS survey notice, in the browser console, type:

```
dispatch( showNpsSurveyNotice() );
```

mattsherman commented:
@shaunandrews I used your original mock as the basis for the styling. Please review and let me know if there are any tweaks you'd like to see!

a8ci18n commented on client/blocks/nps-survey/index.jsx:
```diff
@@ -71,45 +75,51 @@ class NpsSurvey extends Component {
-					Thanks for providing your feedback!
-					<div>
+					<div className="nps-survey__thank-you">
+						{ translate( 'Thanks for your feedback!' ) }
```
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Thanks for your feedback.', { context: 'calypso stats survey'} )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

a8ci18n commented on client/blocks/nps-survey/index.jsx:
```diff
@@ -71,45 +75,51 @@ class NpsSurvey extends Component {
 							onClick={ this.handleFinishClick }
 						>
-							Finish
+							{ translate( 'Finish', { context: 'verb' } ) }
```
:disappointed: 68 existing translations will be lost with this change.
If the string must change, the following option has already been translated **63** times:
`translate( 'Finish', { context: 'spellcheck'} )` *ES Score: 16*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

shaunandrews commented:
<img width="562" alt="screen shot 2017-03-30 at 4 20 19 pm" src="https://cloud.githubusercontent.com/assets/191598/24524511/ced2fb78-1564-11e7-8906-73f5c9de3959.png">

I know others have looked at this copy and made suggestions, but the wording feels very awkward. It doesn't feel natural. I think its the "Would you" part that I'm tripping up on. Maybe something like "Would you mind answering a question..."

--

<img width="375" alt="screen shot 2017-03-30 at 4 11 41 pm" src="https://cloud.githubusercontent.com/assets/191598/24524154/9864f146-1563-11e7-90f0-a7bb1c2f2c2d.png">

Looking at the "Unlikely" and "Very Likely" labels again, I wonder if they could be confusing — they don't visually line up with a specific radio button. Maybe we can add some visual cue to help associate the labels with 0 and 10:

![default state](https://cloud.githubusercontent.com/assets/191598/24524216/c3704d9a-1563-11e7-855b-c439ba8a4458.png)

--

I also notice the empty space where the button will appear:

<img width="544" alt="screen shot 2017-03-30 at 4 13 34 pm" src="https://cloud.githubusercontent.com/assets/191598/24524295/01ba3afc-1564-11e7-9378-e2b8a4a93316.png">

Lets clean this up in the default state; Once they select a radio the button can transition in:

![nps survey button transition](https://cloud.githubusercontent.com/assets/191598/24524489/bd5c402a-1564-11e7-9a9e-3cbc23e6c84f.gif)

--

The success page is a little underwhelming. I wonder if instead, we close the modal immediately and show a success notice:

![success notice](https://cloud.githubusercontent.com/assets/191598/24524694/89700eee-1565-11e7-84b6-9fec832075d4.png)

mattsherman commented:
Thanks for the feedback @shaunandrews -- I will work on incorporating those changes.

I'm not sure about the visual cue for the "Unlikely" and "Likely" labels -- I don't think the visual cues are necessary, though I do see where there could be confusion. I'll give it a try though!

I like the success notice idea instead of the success page.

mattsherman commented:
@shaunandrews The horizontal spacing is already extremely tight on mobile. I don't think it can accommodate the visual cues for the "Unlikely" and "Very Likely" labels, as it pushes those labels out further horizontally.

<img width="338" alt="screencapture at fri mar 31 07 37 34 edt 2017" src="https://cloud.githubusercontent.com/assets/2098816/24548945/1c7b25b4-15e5-11e7-900e-a86c9d6b7e5e.png">

a8ci18n commented on client/blocks/nps-survey/index.jsx:
```diff
@@ -49,49 +51,57 @@ class NpsSurvey extends Component {
 							onClick={ this.handleFinishClick }
 						>
-							Finish
+							{ translate( 'Finish', { context: 'verb' } ) }
```
:disappointed: 68 existing translations will be lost with this change.
If the string must change, the following option has already been translated **63** times:
`translate( 'Finish', { context: 'spellcheck'} )` *ES Score: 16*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/nps-survey-styles)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

yoavf replied:
👋 
You can use `Finish` with no context. It's mostly not ambiguous, and we've already got that translated.
(The a8ci18n bot comment here is somewhat broken, but you can see the proper suggestion here: https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/nps-survey-styles)

mattsherman replied:
Thanks for the clarification @yoavf -- I'll drop the context.

a8ci18n commented on client/blocks/nps-survey/index.jsx:
```diff
@@ -38,75 +40,83 @@ class NpsSurvey extends Component {
+
+	showThanksNotice = () => {
+		this.props.successNotice(
+			this.props.translate( 'Thanks for your feedback!' ),
```
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Thanks for your feedback.', { context: 'calypso stats survey'} )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

mattsherman commented:
@shaunandrews I have changed the following, based on your feedback:

- Notice prompt text
- Switched from thank you screen to success notice
- Finish button transition (I slightly differed in the transition from what you had mocked, because the dialog re-centers when it changes height, and that felt odd. So, I implemented it such that the dialog maintains a constant height.

I did not end up changing the "Unlikely" and "Very Likely" labels, because:

- On mobile there would be horizontal space issues
- With translations, there would be layout issues, as if the translations were much longer, things would look odd

We can iterate further on these labels if desired in a future PR after this one is merged.

/cc @mjuhasz @jonathansadowski @jblz This is is now ready for review.

jonathansadowski commented:
> I know others have looked at this copy and made suggestions, but the wording feels very awkward. It doesn't feel natural. I think its the "Would you" part that I'm tripping up on. Maybe something like "Would you mind answering a question..."

The problem with the question "Would you mind" is that "sure" is technically a response that means "leave me alone, I mind you bugging me".

What about "Have a second to answer one question about WordPress.com?"?

mattsherman commented:
Thanks for the review @jonathansadowski -- I agree that we can do further wording tweaks in a follow-up PR, since this feature isn't live yet.

mjuhasz commented:
Good job 👍

Jon tested it in Chrome (desktop and mobile), Firefox, and IE11, and I can confirm that it works well in Microsoft Edge and Safari as well.
